### ISSUE TITLE
Sik-Ho Tang | Review: PReLU-Net -- The First to Surpass Human-Level Performance in ILSVRC 2015 (Image Classification).

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#92]
Sik-Ho Tang. [Review: PReLU-Net — The First to Surpass Human-Level Performance in ILSVRC 2015 (Image Classification)](https://medium.com/coinmonks/review-prelu-net-the-first-to-surpass-human-level-performance-in-ilsvrc-2015-image-f619dddd5617).

**Comment by NorbertZheng** [comment]
## Overview
In this story, PReLU-Net [1] is reviewed. Parametric Rectified Linear Unit (PReLU) is proposed to generalize the traditional rectified unit (ReLU). This is the deep learning approach which is first to surpass the human-level performance in ILSVRC (ImageNet Large Scale Visual Recognition Challenge) image classification. In addition, **a better weight initialization for rectifiers is proposed which can help with the convergence of deep model (30 layers) trained directly from scratch**.

Finally, PReLU-Net obtains 4.94% top-5 error rate on test set which is better than the human-level performance of 5.1%, and GoogLeNet of 6.66% !!!

And this is the paper in 2015 ICCV and it got about 3000 citations at the moment I am writing this story.

**Comment by NorbertZheng** [comment]
## Dataset
Classification: Over 15 millions labeled high-resolution images with around 22,000 categories. ILSVRC uses a subset of ImageNet of around 1000 images in each of 1000 categories. In all, there are roughly 1.3M/50k/100k images are used for the training/validation/testing sets.

**Comment by NorbertZheng** [comment]
## Parametric Rectified Linear Unit (PReLU)
In AlexNet [2], ReLU is suggested as below where only positive values would pass through the ReLU activation function while all negative values are set to zero. And ReLU outperform Tanh with much faster training speed due to non-saturation at 1.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/d2dacc78-f2b4-4a62-9fb0-8841d468f7a7)
PReLU (LeakyReLU?)

 - PReLU is suggested that there should be penalty for negative values and it should be parametric.

It is noted that when a = 0, it is ReLU. When a = 0.01, it is Leaky ReLU. **Now the value of a can be learnt, therefore becoming a generalized ReLU.**

During backpropagation, we can estimate the gradient:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/234f4655-fb6a-43f4-8f95-6ad96372a97a)
Backpropagation, gradient from deep layer (Left), gradient of the activation (Right).

We can estimate the gradient from the deep layer (left), and the gradient of the activation (right). We can see that it is the sum of all positions of the feature map (Channel-wise). If it is channel-shared variant, it is the sum all over the channels of the layer. No weight decay is applied to a.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ba97a4d9-44c3-4139-9c61-2056475cf37b)
The average value of a over all channels for each layer.

Two interesting phenomena observed:
 - First, the first conv layer (conv1) has coefficients (0.681 and 0.596) significantly greater than 0. As the filters of conv1 are mostly Gabor-like filters such as edge or texture detectors, the learned results show that both positive and negative responses of the filters are respected.
 - For the channel-wise version, the deeper conv layers in general have smaller coefficients. the activations gradually become “more nonlinear” at increasing depths. In other words, the learned model tends to keep more information in earlier stages and becomes more discriminative in deeper stages.

**Comment by NorbertZheng** [comment]
## A better weight initialization for rectifiers
 - A good weight initialization is essential to not to let the network to reduce or magnifying the input signals exponentially.

Since weight initialization depends on Gaussian distribution with mean 0 and different variance depending on the algorithm like Xavier. Thus, by considering the input and output network sizes, a better weight initialization is suggested.

With $L$ layers putting together, the variance is (left):
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a8f0f740-c318-45a2-8fcc-0ae4194e98c3)
Variance of L layers (Left), and sufficient condition (right).

If the sufficient condition at the right is met, the network can become stable. Thus, finally, the variance should be $\frac{2}{n_{l}}$ where $n_{l}$ is the number of connection in $l$-th layer.

(There is also proof at backward propagation case. It is also interesting that **it also comes up with the same sufficient condition**. But I just not to show it here.)

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/594e2bd6-58ff-40ab-b43d-739701edb6a7)
Red (Ours) and Blue (Xavier), 22-layer (Left) and 30-layer (Right).

As shown above, the suggested weight initialization converges faster. And Xavier even cannot converge for the deeper layer at the right when training from scratch.

**Comment by NorbertZheng** [comment]
## 22-layer deep learning models
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a5585720-11ee-454e-b5c3-7ca713257579)
PReLU-Net: Model A, B, C.

 - SPP layer: 4-level SPPNet [3–4] {7×7, 3×3, 2×2, 1×1}
 - Model A: A model with better result than VGG-19 [5]
 - Model B: Deep model than Model A
 - Model C: Wider model (More filters) than Model B

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/61974b73-d2fd-46bc-aff6-17652e6406e5)
Model A using PReLU is better than the one using ReLU.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/cb5f5b6d-6216-4450-a3e8-c10a869b61f4)
Model A: PReLU converges faster.

**Comment by NorbertZheng** [comment]
## Comparison with state-of-the-art approaches
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/f3646dfb-cb8c-4de0-bdf8-074bd1c49e2f)
Single model, 10-view.

By using just single model and 10-view (data augmentation), Model C has 7.38% error rate.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/26173c91-aa6b-4827-8e0b-e23bde6eb059)
Single model, Multi-view, Multi-scale.

With multi-view and multi-scale, Model C has 5.71% error rate. This result is already better than even the multi model of SPPNet [3–4], VGGNet [5] and GoogLeNet [6].

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/7997ec9d-12d5-45cf-a271-04f3b908e0b0)
Multi-model, Multi-view, Multi-scale.

With multi-model, i.e. 6 models PReLU-Net, got the 4.94% error rate.
This is 26% relative improvement against GoogLeNet!!!

**Comment by NorbertZheng** [comment]
## Object detection by using Fast R-CNN
PReLU-Net uses Fast R-CNN [7] implementation for object detection in PASCAL VOC 2007 dataset.
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/1ba9c86a-d9ee-4df2-a301-106885f88f0e)
Model C + PReLU-Net has the best mAP result.

With imagenet pretrained model and fine-tuning on VOC 2007 dataset, Model C obtains better results than VGG-16.

**Comment by NorbertZheng** [comment]
## References
 - [2015 ICCV] [PReLU-Net] [Delving Deep into Rectifiers: Surpassing Human-Level Performance on ImageNet Classification](https://arxiv.org/pdf/1502.01852).
 - [2012 NIPS] [AlexNet] [ImageNet Classification with Deep Convolutional Neural Networks](https://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf).
 - [2014 ECCV] [SPPNet] [Spatial Pyramid Pooling in Deep Convolutional Networks for Visual Recognition](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.699.8052&rep=rep1&type=pdf).
 - [2015 TPAMI] [SPPNet] [Spatial Pyramid Pooling in Deep Convolutional Networks for Visual Recognition](https://arxiv.org/pdf/1406.4729.pdf).
 - [2015 ICLR] [VGGNet] [Very Deep Convolutional Networks for Large-Scale Image Recognition](https://arxiv.org/pdf/1409.1556).
 - [2015] [CVPR] [GoogLeNet] [Going Deeper with Convolutions](https://www.cs.unc.edu/~wliu/papers/GoogLeNet.pdf).
 - [2015 ICCV] [Fast R-CNN] [Fast R-CNN](https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/Girshick_Fast_R-CNN_ICCV_2015_paper.pdf).
